### PR TITLE
feat: LOOP and ASTROPORT support on router

### PR DESCRIPTION
**Pull request by JoowonYun**
## Summary of changes

<Describe summary of major changes here>

## Report of required housekeeping

- [x] Github issue OR spec proposal link
- [x] Wrote tests
- [x] Added a relevant changelog entry to CHANGELOG.md

----

## (FOR ADMIN) Before merging

- [ ] Added appropriate labels to PR
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" (coding standards)
- [ ] Confirm added tests are consistent with the intended behavior of changes
- [ ] Ensure all tests pass

### CODE REVIEW

**Comment by JoowonYun** [comment]
Pending https://github.com/terraswap/terraswap/pull/30

File: contracts/terraswap_router/src/contract.rs [hunk 1]
```diff
@@ -317,6 +330,154 @@ fn simulate_swap_operations(
...
+    let mut res = reverse_simulate(
+        &deps.querier,
+        Addr::unchecked(pair_info.contract_addr),
+        &Asset {
+            amount: ask_amount,
+            info: ask_asset_info,
+        },
+    )?;
+
+    // Deduct tax after querying simulation
```
Comment (jhlee-young) [inline contracts/terraswap_router/src/contract.rs:469]: needs update comment 'deduct' to 'add'?

File: contracts/terraswap_router/src/state.rs [hunk 2]
```diff
@@ -7,6 +7,8 @@ use cw_storage_plus::Item;
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 pub struct Config {
     pub terraswap_factory: CanonicalAddr,
+    pub loop_factory: CanonicalAddr,
```
Comment (jhlee-young) [inline contracts/terraswap_router/src/state.rs:10]: do we need schema json update as well?

Comment (JoowonYun) [reply contracts/terraswap_router/src/state.rs:10]: https://github.com/terraswap/terraswap/pull/34/commits/7d56451c033fb07288eda67115e31a6742c76e00

File: contracts/terraswap_router/src/contract.rs (hunk 1 above)
Comment (JoowonYun) [reply contracts/terraswap_router/src/contract.rs]: https://github.com/terraswap/terraswap/pull/34/commits/7d56451c033fb07288eda67115e31a6742c76e00

**Review by YunSuk-Yeo — APPROVED** [review]

**Review by jhlee-young — APPROVED** [review]

**Pull request merged**